src/controller/CHIPDeviceController.cpp

```diff
@@ -731,7 +731,8 @@ CHIP_ERROR DeviceCommissioner::EstablishPASEConnection(NodeId remoteDeviceId, Re
     exchangeCtxt = mSystemState->ExchangeMgr()->NewContext(session.Value(), &device->GetPairing());
     VerifyOrExit(exchangeCtxt != nullptr, err = CHIP_ERROR_INTERNAL);
 
-    err = device->GetPairing().Pair(*mSystemState->SessionMgr(), params.GetSetupPINCode(), GetLocalMRPConfig(), exchangeCtxt, this);
+    err = device->GetPairing().Pair(*mSystemState->SessionMgr(), params.GetSetupPINCode(), GetLocalMRPConfig(), exchangeCtxt, this,
+                                    mSystemState->GetSessionKeystore());
```

Comment:
Really feels like Pair took too many arguments before, and it's more of a problem now. Any way to avoid passiong Keystore and SessionMgr down by doing some factoring on the class with that method?